tests/frontend/org/voltdb/regressionsuites/TestCalciteAdHocPlannerCache.java

```
@@ -0,0 +1,172 @@
+    }
+
+    public void testCalciteCacheHitMiss() throws IOException, ProcCallException {
+        System.out.println("testAdHocCalcitePlannerCache...");
+        resetStatistics();
+        Client client = getClient();
+        String sql = "SELECT * FROM R1 WHERE ID=1";
+
+        client.callProcedure("R1.insert", 1, "foo1", 0, 1.1);
+        client.callProcedure("R1.insert", 2, "foo2", 0, 2.2);
+        client.callProcedure("R1.insert", 3, "foo3", 1, 3.3);
+
+        // Test cache miss
+        client.callProcedure("@AdHoc", sql);
+        checkCalcitePlannerCache(client, Calcite_CACHE_MISS2);
+
+        // Test hit L1 cache
+        client.callProcedure("@AdHoc", sql);
+        checkCalcitePlannerCache(client, Calcite_CACHE_HIT1);
+
+        // Test hit L2 cache
+        client.callProcedure("@AdHoc", sql.substring(0, 26) + "2");
+        checkCalcitePlannerCache(client, Calcite_CACHE_HIT2);
+    }
+    //
+    // Suite builder boilerplate
+    //
+
+    public TestCalciteAdHocPlannerCache(String name) {
+        super(name);
+    }
+    static final Class<?>[] PROCEDURES = {
+        org.voltdb_testprocs.regressionsuites.plansgroupbyprocs.CountT1A1.class,
+        org.voltdb_testprocs.regressionsuites.plansgroupbyprocs.SumGroupSingleJoin.class };
+
+    static public junit.framework.Test suite() {
+        VoltServerConfig config;
+        MultiConfigSuiteBuilder builder = new MultiConfigSuiteBuilder(
+                TestCalciteAdHocPlannerCache.class);
+        VoltProjectBuilder project = new VoltProjectBuilder();
+        final String literalSchema =
+                "CREATE TABLE R1 ( " +
+                " ID BIGINT DEFAULT 0 NOT NULL, "
+                + "DESC VARCHAR(300), "
+                + "NUM bigint,"
+                + "RATIO FLOAT, "
+                + "PRIMARY KEY (desc)); ";
+        try {
+            project.addLiteralSchema(literalSchema);
+        } catch (IOException e) {
+            assertFalse(true);
```

why not just let exception propagate?

---

I think this implementation is to assert failure without interrupting the test case

---

updated, thx